dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -172,7 +158,7 @@ check_not_exporting_variables() {
     # Only run if there are patterns, otherwise it matches everything
     if [ ${#patterns[@]} -ne 0 ]; then
         git -C "${RIOTBASE}" grep -n "${patterns[@]}" -- "${pathspec[@]}" \
-            | error_with_message 'Variables must only be exported in `makefiles/vars.inc.mk`'
+            | error_with_message "Variables must only be exported in \`makefiles/vars.inc.mk\`"
```

Comment:
Mh... Are we sure we want to introduce escapes instead of just using `'`s? If shellcheck is complaining about this for some reason, I'd handle this as a false positive and suppress the warning here.

Reply:
The thing is that if you see backticks or `"$(cmd)"` somewhere in a string, you may be lead to the conclusion that a shell command is executed until you notice that it is a `'` string and not a `"` string (both usually have the same color highlighting).

Escaping those makes it obvious to a reviewer that the backticks have no special meaning. Hence, I do believe that shellcheck does the right thing here.